OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -141,15 +147,20 @@ public Armament(Actor self, ArmamentInfo info)
 
 		public virtual WDist MaxRange()
 		{
-			return new WDist(Util.ApplyPercentageModifiers(Weapon.Range.Length, rangeModifiers));
+			return new WDist(Util.ApplyPercentageModifiers(Weapon.Range.Length, initialRangeModifiers));
```

Comment:
Uh, I might be missing something here, but to my understanding the old code didn't cache the traits at all, it only cached the modifiers once at Creation (which I'm still doing here) and then looked them up anew every time in (old) lines 237+238 further below.